src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenRefReturnTests.vb

```diff
@@ -2577,6 +2577,443 @@ End Module
+        <WorkItem("https://github.com/dotnet/roslyn/issues/68194")>
+        Public Sub With_Method_01()
+            Dim comp1 = CreateCSharpCompilation(
+"public class C<T>
+{
+#pragma warning disable 0649
+    private T _p;
+    public ref T GetP()
+    {
+        return ref _p;
+    }
+}")
+            comp1.VerifyDiagnostics()
+            Dim comp2 = CreateVisualBasicCompilation(
+                Nothing,
+"Module M
+    Sub Main()
+        Dim o = New C(Of Integer)()
+        o.GetP() += 1
+        With o.GetP()
+            System.Console.Write(.ToString())
+            o.GetP() = 2
+            System.Console.Write(.ToString())
+        End With
+    End Sub
+End Module",
+                referencedCompilations:={comp1},
+                compilationOptions:=TestOptions.DebugExe)
+            Dim verifier = CompileAndVerify(comp2, expectedOutput:="12")
+            verifier.VerifyIL("M.Main",
+            <![CDATA[
+{
+  // Code size       62 (0x3e)
+  .maxstack  3
+  .locals init (C(Of Integer) V_0, //o
+                Integer& V_1,
+                Integer& V_2) //$W0
+  IL_0000:  nop
+  IL_0001:  newobj     "Sub C(Of Integer)..ctor()"
+  IL_0006:  stloc.0
+  IL_0007:  ldloc.0
+  IL_0008:  callvirt   "ByRef Function C(Of Integer).GetP() As Integer"
+  IL_000d:  dup
+  IL_000e:  stloc.1
+  IL_000f:  ldloc.1
+  IL_0010:  ldind.i4
+  IL_0011:  ldc.i4.1
+  IL_0012:  add.ovf
+  IL_0013:  stind.i4
+  IL_0014:  nop
+  IL_0015:  ldloc.0
+  IL_0016:  callvirt   "ByRef Function C(Of Integer).GetP() As Integer"
+  IL_001b:  stloc.2
+  IL_001c:  ldloc.2
+  IL_001d:  call       "Function Integer.ToString() As String"
+  IL_0022:  call       "Sub System.Console.Write(String)"
+  IL_0027:  nop
+  IL_0028:  ldloc.0
+  IL_0029:  callvirt   "ByRef Function C(Of Integer).GetP() As Integer"
+  IL_002e:  ldc.i4.2
+  IL_002f:  stind.i4
+  IL_0030:  ldloc.2
+  IL_0031:  call       "Function Integer.ToString() As String"
+  IL_0036:  call       "Sub System.Console.Write(String)"
+  IL_003b:  nop
+  IL_003c:  nop
+  IL_003d:  ret
+}
+]]>)
+            verifier.VerifyDiagnostics()
+        End Sub
+
+        <Fact>
+        <WorkItem("https://github.com/dotnet/roslyn/issues/68194")>
+        Public Sub With_Method_02()
+            Dim comp1 = CreateCSharpCompilation(
+"public class C<T>
+{
+#pragma warning disable 0649
+    private T _p;
+    public ref T GetP()
+    {
+        return ref _p;
+    }
+}")
+            comp1.VerifyDiagnostics()
+            Dim comp2 = CreateVisualBasicCompilation(
+                Nothing,
+"
+#disable warning BC42356
+
+Module M
+    async Sub Main()
+        Dim o = New C(Of Integer)()
+        With o.GetP()
+        End With
+    End Sub
+End Module",
+                referencedCompilations:={comp1},
+                compilationOptions:=TestOptions.DebugDll)
+            comp2.AssertTheseDiagnostics(
+<expected>
+BC37326: A call to a method or property that returns by reference may not be used in this context.
```

Comment:
>this context. [](http://example.com/codeflow?start=85&length=13)

nit: I had a bit of trouble understanding this error until I noticed `async`. Consider adding more details in the error to give a clue about what context is okay or not.